test/extensions/filters/network/http_connection_manager/config_test.cc

```diff
@@ -1850,6 +1850,7 @@ stat_prefix: router
     config_source: { resource_api_version: V3, ads: {} }
     default_config:
       "@type": type.googleapis.com/envoy.extensions.filters.http.health_check.v3.HealthCheck
+      pass_through_mode: false
```

Comment:
Why this change?

Reply:
ref the HealthCheck proto:
https://github.com/envoyproxy/envoy/blob/5b1977f607962d7b700ce486651ca12bd5659851/api/envoy/extensions/filters/http/health_check/v3/health_check.proto#L31-L32, it requires `pass_through_mode` to be set. Otherwise, it will raise validate error but not terminal filter not fit error.

Reply:
For that to be the case won't the HealthCheck extension's isTerminalFilter function need to be updated in this PR to read the config? I don't see a change like that, but maybe I'm missing something?

Reply:
Most HTTP filter class is inheriting from `Common::FactoryBase<T>`, and use the default isTerminalFilter function from the parent, I changed it at this class.

Reply:
I don't see you changing the the function for the HealthCheck filter? Shouldn't it be using the default FactoryBase one that always returns false?

It seems right to me that the HealthCheck filter should change isTerminal based on this flag, I just don't see that code included in this PR

Reply:
1. I changed the `isTerminalFilter` function at the base class `Common::FactoryBase<T>`, which is the base class of HealthCheck filter factory. https://github.com/qinggniq/envoy/blob/b9dfe5c0a780ca8d08230aa9aeace6e3e6cf4c47/source/extensions/filters/http/common/factory_base.h#L45-L50
2. I change the validating sequence from `terminal filter validate -> parse protocol message (validate message) -> create filter` to `parse protocol message (validate message)  -> terminal filter validate  -> create filter` because now isTerminalFilter rely on message.
3. I added `pass_through_mode = 1` because after point 2 I mentioned, the previous validate sequence will not parse the message because it will raise *terminal filter to validate error* first, after change the validating sequence,  it will raise `validate message error` first, which it's not we want to test.

Do these clarify clear?

Reply:
Oh I see, this is about message validation, not about it affecting terminal filters. Thanks for explaining!